posts/copilot-workspace/index.qmd

```diff
@@ -0,0 +1,93 @@
+
+Given the rapid development and increasing availability to artificial intelligence (AI) models, specifically large language models, offer the potential to be another development tool to add to the arsenal of researchers and software engineers to enhance productivity and improve the reliability of the code we write and others depend upon [@vanHoek2024]. Large language models like ChatGPT from OpenAI, Gemini from Google, Claude from Anthropic, as well as many others and new ones frequently appearing, have presented opportunities to quickly generate material -- text, code and images -- from text prompts.
+
+We decided to explore the suitability of one such, GitHub Copilot integrated with GitHub Workspaces, for some typical use cases encountered by our research software engineering team. We do note that many other such tools are available, and we chose Copilot in Workspaces as our code is hosted on GitHub, making for easy evaluation of the specific contribution of the AI tool. However, some of our broad conclusions are based on using other tools for similar tasks.
+
+To test this out, a small group of Epiverse-TRACE professors and research software engineers explored its functionality. The aim of this blog post is to summarise our experience with Copilot and Workspaces, and share our thoughts more generally about how the current feature set of AI-powered development tools could help with our daily development tasks, and with research software development at large. We evaluated Copilot workspace across three tasks of varying difficulty and requirements:
+
+* Experiment 1: Export an R function by updating the documentation
+
+* Experiment 2: Adding a basic Susceptible-Infected-Recovered (SIR) model to an existing R package that contains other epidemiological models
+
+* Experiment 3: Conducting a code review on an R package
+
+## Experiment 1: Export an R function by updating the documentation
+
+_Difficulty level: basic_
+
+The first experiment was on an R package {EpiNow2} used for forecasting and nowcasting epidemics and pandemics. [Issue #681](https://github.com/epiforecasts/EpiNow2/issues/681) was chosen as a simple task to evaluate Copilot workspace's ability to interpret requests that require small changes to the codebase of a package (i.e. can be achieved in a single commit by changing less than 5 lines of code). The issue stated that an internal function, `epinow2_cmdstan_model()` should be exported, with the issue description explaining this would help users specify compile-time options.
```

Comment:
```suggestion
The first experiment was on an R package {EpiNow2}, which provides functions for forecasting and nowcasting epidemics and pandemics. [Issue #681](https://github.com/epiforecasts/EpiNow2/issues/681) was chosen as a simple task to evaluate Copilot workspace's ability to interpret requests that require small changes to the codebase of a package (i.e. can be achieved in a single commit by changing less than 5 lines of code). The issue stated that an internal function, `epinow2_cmdstan_model()` should be exported, with the issue description explaining this would help users specify compile-time options.
```